### PR TITLE
[wx] Improve context menu handling at tree view

### PR DESCRIPTION
**Pull request by rafaelx**
Resolves reported issue [1217](https://sourceforge.net/p/passwordsafe/bugs/1217/).

### CODE REVIEW

File: src/ui/wxWidgets/TreeCtrl.cpp [hunk 1]
```diff
@@ -99,6 +101,24 @@ BEGIN_EVENT_TABLE( TreeCtrl, wxTreeCtrl )
...
+    In Linux environments context menus appear on Right-Down mouse click.
+    Which mouse click type (Right-Down/Right-Up) is the right one on a platform
+    is considered by 'EVT_CONTEXT_MENU(TreeCtrl::OnContextMenu)' which doesn't
+    work for wxTreeCtrl prior wx version 3.1.1. See also the following URL.
+    https://github.com/wxWidgets/wxWidgets/commit/caea08e6b2b9e4843e84c61abc879880a08634b0
+  */
+#if wxCHECK_VERSION(3, 1, 1)
+  EVT_CONTEXT_MENU(TreeCtrl::OnContextMenu)
+#else
+#ifdef __WINDOWS__
```
Comment (dskrvk) [inline src/ui/wxWidgets/TreeCtrl.cpp]: Isn't the whole wxWidgets part Linux-specific?

File: src/ui/wxWidgets/TreeCtrl.cpp [hunk 2]
```diff
@@ -785,6 +819,46 @@ void TreeCtrl::OnKeyDown(wxTreeEvent& evt)
...
+}
+
+void TreeCtrl::OnMouseLeftClick(wxMouseEvent& event)
+{
+  wxPoint mouseClickPosition = event.GetPosition();
+  int positionInfo;
+
+  HitTest(mouseClickPosition, positionInfo);
+
+  if ((positionInfo | wxTREE_HITTEST_NOWHERE) == wxTREE_HITTEST_NOWHERE) {
```
Comment (dskrvk) [inline src/ui/wxWidgets/TreeCtrl.cpp]: I don't know much about the wxWidgets API, but if `wxTREE_HITTEST_NOWHERE` is a mask with a single bit set to 1, shouldn't the operator be `&`?

File: src/ui/wxWidgets/TreeCtrl.cpp (hunk 1 above)
Comment (ronys) [reply src/ui/wxWidgets/TreeCtrl.cpp]: Actually, no.
- The wx code also runs on Mac
- It's also built and run under Windows as an exercise in portability.

File: src/ui/wxWidgets/TreeCtrl.cpp (hunk 2 above)
Comment (ronys) [reply src/ui/wxWidgets/TreeCtrl.cpp]: Agree - certainly looks like this should be "&", not "|"

File: src/ui/wxWidgets/TreeCtrl.cpp (hunk 1 above)
Comment (rafaelx) [reply src/ui/wxWidgets/TreeCtrl.cpp]: Would be great if someone could test the implementation with wxWidgets 3.1.1 and on platforms other than Linux.

File: src/ui/wxWidgets/TreeCtrl.cpp (hunk 2 above)
Comment (rafaelx) [reply src/ui/wxWidgets/TreeCtrl.cpp]: Ow, good catch. - Fixed.

Comment (dskrvk) [reply src/ui/wxWidgets/TreeCtrl.cpp]: Fixed in OnMouseRightClick but not in OnMouseLeftClick.

Comment (rafaelx) [reply src/ui/wxWidgets/TreeCtrl.cpp]: Thanks for pointing this out. Fixed in #609.

**Pull request merged**